src/pipx/util.py

```diff
@@ -98,11 +93,18 @@ def run_subprocess(
     capture_stderr: bool = True,
 ) -> subprocess.CompletedProcess:
     """Run arbitrary command as subprocess, capturing stderr and stout"""
+    env = dict(os.environ)
 
-    # Null out PYTHONPATH because some platforms (macOS with Homebrew) add
+    # Remove PYTHONPATH because some platforms (macOS with Homebrew) add
     #   pipx directories to it, and can make it appear to venvs as though
     #   pipx dependencies are in the venv path (#233)
-    env = {k: v for k, v in os.environ.items() if k.upper() != "PYTHONPATH"}
+    # Remove __PYVENV_LAUNCHER__ because it is only meant to be communication
+    #   between main python binary and its subprocess and can cause the wrong
+    #   python binary to be used (#334)
```

Comment:
nit: this is a little confusing. Maybe change to
```
    # Remove __PYVENV_LAUNCHER__ because it can cause the wrong
    #   python binary to be used (#334)
```

Reply:
Done